src/ngraph/op/embedding_segments_sum.cpp

```diff
@@ -0,0 +1,199 @@
+                              get_input_partial_shape(SEGMENT_IDS).to_shape().size() == 1,
+                          "SEGMENT_IDS must be 1D");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_partial_shape(INDICES).compatible(get_input_partial_shape(SEGMENT_IDS)),
+        "INDICES and SEGMENT_IDS shape must be same");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_partial_shape(NUM_SEGMENTS).compatible(PartialShape{}),
+                          "NUM_SEGMENTS must be a scalar");
+
+    if (get_input_size() >= 5)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(DEFAULT_INDEX) == element::i64 ||
+                                  get_input_element_type(DEFAULT_INDEX) == element::i32,
+                              "DEFAULT_INDEX type must be i32 or i64");
+
+        NODE_VALIDATION_CHECK(
+            this,
+            get_input_element_type(INDICES).compatible(get_input_element_type(DEFAULT_INDEX)),
+            "Default_index element type (",
+            get_input_element_type(DEFAULT_INDEX),
+            ") must match indices element type (",
+            get_input_element_type(INDICES),
+            ")");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(DEFAULT_INDEX).compatible(PartialShape{}),
+                              "DEFAULT_INDEX must be a scalar");
+    }
+
+    if (get_input_size() == 6)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(EMB_TABLE).compatible(
+                                  get_input_element_type(PER_SAMPLE_WEIGHTS)),
+                              "Per sample weight element type (",
+                              get_input_element_type(PER_SAMPLE_WEIGHTS),
+                              ") must match embedding table element type (",
+                              get_input_element_type(EMB_TABLE),
+                              ")");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(PER_SAMPLE_WEIGHTS).is_dynamic() ||
+                                  get_input_partial_shape(PER_SAMPLE_WEIGHTS).to_shape().size() ==
+                                      1,
+                              "PER_SAMPLE_WEIGHTS must be 1D");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(INDICES).compatible(
+                                  get_input_partial_shape(PER_SAMPLE_WEIGHTS)),
+                              "INDICES and PER_SAMPLE_WEIGHTS shape must be same");
+    }
+
+    element::Type result_et = get_input_element_type(EMB_TABLE);
+
+    const PartialShape& emb_table_shape = get_input_partial_shape(EMB_TABLE);
+
+    PartialShape result_shape;
+    if (emb_table_shape.rank().is_static())
+    {
+        result_shape = emb_table_shape;
+        result_shape[0] = Dimension::dynamic();
```

Comment:
Please, check that the input with `num_segments` is Constant and if so use to set the `result_shape[0]`